src/librbd/plugin/Types.h

```diff
@@ -14,8 +15,43 @@ namespace plugin {
 
 template <typename> struct Api;
 
+template <typename ImageCtxT>
+struct HookPointInterface {
+  virtual void open(ImageCtxT* image_ctx, Context* on_finish) = 0;
+  virtual void close(ImageCtxT* image_ctx, Context* on_finish) = 0;
+  virtual void discard(ImageCtxT* image_ctx, Context* on_finish) {
+    on_finish->complete(0);
+  }
+
+  virtual ~HookPointInterface() {};
+};
+
+template <typename ImageCtxT>
 struct HookPoints {
```

Comment:
Nit: no need for extra wrapper around interface.

Reply:
DONE